os/kernel/log_dump/log_dump_save.c

```diff
@@ -0,0 +1,171 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/log_dump/log_dump_internal.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/****************************************************************************
+ * Private Type Declarations
+ ****************************************************************************/
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+/****************************************************************************
+ * Private Variables
+ ****************************************************************************/
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int __exception(void)
+{
+	if (IS_KMM_LOCKED())
+		return LOG_DUMP_MEM_FAIL;
+	if (sched_self()->sched_priority > CONFIG_LOG_DUMP_MANAGER_PRIO)
+		return LOG_DUMP_FAIL;
+	if (sched_lockcount())
+		return LOG_DUMP_FAIL;
+	return LOG_DUMP_OK;
+}
+
+static int __request_compress(const int index)
+{
+	struct log_dump_msg comp_msg;
+	comp_msg.uncomp_idx = index;
+
+	int ret = mq_send(mq_fd, (const char *)&comp_msg, sizeof(struct log_dump_msg), 100);
```

Comment:
For many cases, to debug the failure of mq creation, it is better to check the return value of `mq_send`. 
How about printing if failed?